### ISSUE TITLE
Infinite scroll pagination

### ISSUE THREAD

**Issue by obsidienne** [woylie/flop_phoenix#192]
**Is your feature request related to a problem? Please describe.**
To adapt the pagination to my use case, I need to create a custom component adding an infinite scroll behavior (something with Intersection Observer API probably)

**Describe the solution you'd like**
Create a custom component like cursor_pagination but based on intersection observer.

**Describe alternatives you've considered**
Do it myself 

**Additional context**
Are you interested by a PR or have you already find a solution with the actual capacity ?

**Comment by woylie** [comment]
I didn't implement this yet, but a PR is welcome! Should be based on LiveView streams now.

**Comment by obsidienne** [comment]
Hi, yes LiveView streams and the new phx-viewport-top & phx-viewport-bottom

https://youtu.be/FADQAnq0RpA?t=2373

May be wait for it to ship?

**Comment by woylie** [comment]
Yes, maybe better.

**Comment by Thr3x** [comment]
Any update on this? It would be really nice

**Comment by woylie** [comment]
LiveView has `phx-viewport-top` and `phx-viewport-bottom` attributes now. There's an [example for infinite scroll](https://hexdocs.pm/phoenix_live_view/bindings.html#scroll-events-and-infinite-stream-pagination) in the documentation. All you need to do on the Flop side is to set the pagination type to `:first` or `:last` and call [Flop.to_next_cursor/1](https://hexdocs.pm/flop/Flop.html#to_next_cursor/1) or [Flop.to_previous_cursor/1](https://hexdocs.pm/flop/Flop.html#to_previous_cursor/1) in your event handler.

**Comment by romarickb** [comment]
Hi @Thr3x how you handled it ?
Is it doable with `Flop.Phoenix.table` ?

**Comment by Thr3x** [comment]
@romarickb Yeah I followed @woylie advice using cursor pagination in combination with an anchor. It worked really well.

In my app.js i created a hook for the anchor: 

```
  entries: {
    mounted() {
      observer = new IntersectionObserver((entries) => {
        entries.forEach(entry => {
          if (entry.isIntersecting) {
            this.pushEvent("load-more", {});
          }
        });
      }, {
        root: null, // Use the viewport as the root
        rootMargin: '0px',
        threshold: 1.0 // Trigger when the element is fully visible
      });

      const anchor = document.getElementById('entries-anchor');
      if (anchor) {
        observer.observe(anchor);
      } else {
        console.error('Anchor element not found: #entries-anchor');
      }
    }
  },
```

And in the liveview I added the hook and the anchor below my table:
```
<div phx-hook="entries">
  <Flop.Phoenix.table>
    Your content here
  </Flop.Phoenix.table>
 <div id="entries-anchor"></div>
</div>
```

This then pushes the event to the server and can be handled below:

```
@impl true
  def handle_event("load-more", _, socket) do
    flop = Flop.to_next_cursor(socket.assigns.meta)

    case Clients.list_clients(flop) do
      {:ok, {clients, meta}} ->
        {:noreply,
         socket
         |> assign(:meta, meta)
         |> stream(:clients, clients)}

      _ ->
        {:noreply, socket}
    end
  end
```

Remember to modify your Flop Schema to use pagination. This was my setup. It only loads the first 25 entries, if the anchor is visible, it loads the next 25.

```
  @derive {
    Flop.Schema,
    filterable: [:id, , :first_name, :last_name, :birthday],
    sortable: [:id, :first_name, :last_name, :birthday],
    default_limit: 25,
    pagination_types: [:first, :last],
    default_pagination_type: :first,
    default_order: %{order_by: [:first_name, :id], order_directions: [:asc]},
  }
```

This should be everything to get you on the right track. I don't think I missed anything

**Comment by romarickb** [comment]
Awesome ! Thanks I was doing exactly the same king of thing as you `<div id="entries-anchor"></div>`

I misunderstood Woylie advice as I thought it was possible to to it using phx-viewport-top and phx-viewport-bottom as the example provided.

Thank you for your very fast answer 😄

**Comment by woylie** [comment]
Changed the label to "documentation". I don't think this will be a feature of the library, but I'd like to have this as part of the guides.

**Comment by Thr3x** [comment]
Definitely a good idea for the docs as an example! It think it's a pretty normal use case and phx-viewport-bottom doesn't seem to work that well. For me at least

**Comment by woylie** [comment]
I added a guide in #464. I used your JS hook as a base, thank you for posting it! @Thr3x
